File: SEP3_Client/Mediator/Information/LoginOrRegisterPackage.cs
namespace SEP3_Client.Mediator.Information
{
    public class LoginOrRegisterPackage : InformationPackage
    {
        public string Id { get; set; }
        public string Password { get; set; }
        public string Name { get; set; }

        public LoginOrRegisterPackage(string keyword,string idOrName,string password) : base(InformationType.LOGIN,keyword)
        {
            switch (keyword)
            {
                case "login":
                    Id = idOrName;
                    break;
                case "register":
                    Name = idOrName;
                    break;
                default:
                    break;
            }
            Password = password;
        }

        public string GetId() {
            return Id;
        }

        public string GetPassword() {
            return Password;
        }
        
        public string GetName() {
            return Name;
        }
    }
}

File: SEP3_UserSystem/src/main/java/mediator/information/LoginOrRegisterPackage.java
package mediator.information;

public class LoginOrRegisterPackage extends InformationPackage{
    private String Id;
    private String Password;
    private String Name;

    public LoginOrRegisterPackage(String keyword,String id,String password)
    {
        super(InformationType.LOGIN,keyword);
        Id = id;
        Password = password;
    }

    public String getId() {
        return Id;
    }

    public String getPassword() {
        return Password;
    }
}
